drivers/rail/rail_drv.c

```diff
@@ -0,0 +1,798 @@
+            // hold packet so it can be received from user thread context
+            RAIL_HoldRxPacket(rhandle);
+            DEBUG("Rail event rx packet  hold packet \n");
+            dev->netdev.netdev.event_callback((netdev_t *) &dev->netdev, NETDEV_EVENT_ISR);
+            DEBUG("Rail event rx packet  after netdev callback \n");
+        }
+
+    }
+
+    // RAIL_EVENT_RX_PACKET_ABORTED
+    //Occurs when a packet is aborted, but a more specific reason (such as
+    // RAIL_EVENT_RX_ADDRESS_FILTERED) isn't known.
+
+
+    // Occurs when a packet was sent
+    if (event & RAIL_EVENT_TX_PACKET_SENT) {
+        DEBUG("Rail event Tx packet sent \n");
+
+        dev->netdev.netdev.event_callback((netdev_t *) &dev->netdev, NETDEV_EVENT_TX_COMPLETE);
+
+        // TODO set state?
+    }
+
+    //
+    // RAIL_EVENT_TXACK_PACKET_SENT
+    // Occurs when an ack packet was sent.
+
+    if (event & RAIL_EVENT_TX_CHANNEL_BUSY) {
+        DEBUG("Rail event Tx channel busy\n");
+        dev->netdev.netdev.event_callback((netdev_t *) &dev->netdev, NETDEV_EVENT_TX_MEDIUM_BUSY);
+
+        // TODO set state?
+    }
+
+    // Occurs when a transmit is aborted by the user
+    if (event & RAIL_EVENT_TX_ABORTED) {
+        DEBUG("Rail event Tx aborted\n");
+
+        // TODO set state?
+    }
+
+    // RAIL_EVENT_TXACK_ABORTED
+    // Occurs when a transmit is aborted by the user
+
+    //  Occurs when a transmit is blocked from occurring due to having called
+    //  RAIL_EnableTxHoldOff().
+    if (event & RAIL_EVENT_TX_BLOCKED) {
+        DEBUG("Rail event Tx blocked\n");
+
+        // TODO set state?
+    }
+
+
+    // RAIL_EVENT_TXACK_BLOCKED
+    //  Occurs when an ack transmit is blocked from occurring due to having
+    //  called RAIL_EnableTxHoldOff().
+
+
+    // Occurs when the transmit buffer underflows.
+    if (event & RAIL_EVENT_TX_UNDERFLOW) {
+        DEBUG("Rail event Tx underflow - > should not happen\n");
+    }
+
+    // RAIL_EVENT_TXACK_UNDERFLOW
+    // Occurs when the ack transmit buffer underflows.
+
+    // Indicates a Data Request is being received when using IEEE 802.15.4 functionality.
+    if (event & RAIL_EVENT_IEEE802154_DATA_REQUEST_COMMAND) {
+        // TODO what is source match? and why might it be neccessary to filter
+        // here the packet?
+        DEBUG("Rail event ieee 802.15.4 data request command\n");
+        RAIL_IEEE802154_SetFramePending(rhandle);
+
+    }
+
+    // Occurs when the application needs to run a calibration.
+    if (event & RAIL_EVENT_CAL_NEEDED) {
+
+        DEBUG("Rail event calibration needed \n");
+        LOG_INFO("Rail radio transceiver needs a calibration: executed\n");
+
+        RAIL_Status_t ret;
+
+        ret = RAIL_Calibrate(rhandle, NULL, RAIL_CAL_ALL_PENDING);
+
+        DEBUG("calibration done, ret: %d \n", ret);
+
+        assert(ret == RAIL_STATUS_NO_ERROR);
+
+    }
+}
+
+const char *rail_error2str(RAIL_Status_t status)
```

Comment:
Would it make sense to put this behind the `#ifdef DEVELHELP`?

Reply:
hmm it depends I guess. I'll try and the compiler is probably going to tell me, if I need them in the normal code ;)